.github/workflows/ci-schedule-compatibility.yaml

```diff
@@ -11,6 +11,7 @@ jobs:
     if: ${{ github.repository == 'karmada-io/karmada' }}
     runs-on: ubuntu-22.04
     strategy:
+      max-parallel: 2
```

Comment:
Hi, @liangyuanpeng, I am not sure, I remember it also failed when I set 2(sometime).

Is there any way to retry it once failed(with 5/6 max-parallel), the threshold could be 3.

Reply:
> , I remember it also failed when I set 2(sometimes).

It used to contain other errors and now only env setup timeout error, I might think the problem you are having is a test case error.

> Is there any way to retry it once fails (with 5/6 max-parallel), the threshold could be 3.


```yaml

jobs:
  e2e:
  ...

  retry:
    name: retry test
    needs: [ e2e ]
    // waiting for all of e2e job ends and then uses the gh command to try again if necessary.
    
```

I once did something similar but I lost the working record, see https://github.com/karmada-io/karmada/issues/4111#issuecomment-1803544337

> https://github.com/marketplace/actions/retry-action

This would leave our github action job in a nested state, which I would probably avoid.

Reply:
I modified like this: 
https://github.com/jwcesign/karmada/blob/f77fdb98d119cec1238bc0887fc36e4b89922ec3/.github/workflows/ci-schedule-compatibility.yaml#L49C1-L55C37

Waitting for the testing result:
https://github.com/jwcesign/karmada/actions/runs/7525890008/job/20483011518

This format looks fine(not really nested state), what do you think?

Reply:
> I modified it like this:
https://github.com/jwcesign/karmada/blob/f77fdb98d119cec1238bc0887fc36e4b89922ec3/.github/workflows/ci-schedule-compatibility.yaml#L49C1-L55C37

Cool, It's seems easy to understand And the changes are very small.

```yaml
...
      - name: setup e2e test environment
        uses: nick-fields/retry@v2.9.0
        with:
          max_attempts: 3
          timeout_minutes: 20
          command: |
            hack/local-up-karmada.sh
...
```